src/Framework/AspNetCoreAnalyzers/src/Analyzers/RouteEmbeddedLanguage/Infrastructure/RouteStringSyntaxDetector.cs

```diff
@@ -151,6 +165,16 @@ private static bool IsAnyStringLiteral(int rawKind)
 
         for (var node = token.Parent; node != null; node = node.Parent)
         {
+            // Searching for the language comment on all parents is expensive (GetLeadingTrivia) if the token is deeply nested.
+            // An example of this is a string with hundreds of concatenations. Avoid this by checking the last inspected string node.
+            // The last inspected string will be a parent and allows us to exit quickly from a deeply nested tree.
```

Comment:
Alternatively I guess you could somehow limit the number of strings in a concatenation you visit. I imagine it's not necessary to support "route string syntax" on strings consisting of lots of concatenated parts. Or don't support comments anywhere inside the concatenation, just at the top level. Not saying those are better solutions, just some thoughts.

Reply:
Yeah, got it. If this didn't work, then I'd fall back to that to fix. Not needed at the moment.